zkevm-circuits/src/keccak_circuit/keccak_packed_multi.rs

```diff
@@ -246,19 +255,20 @@ impl<F: FieldExt> CellManager<F> {
     ) -> Cell<F> {
         let column_idx = self.rows[row_idx as usize];
         self.rows[row_idx as usize] += 1;
-        self.query_cell_at_pos(meta, row_idx, column_idx)
+        self.query_cell_at_pos(meta, row_idx, column_idx, FirstPhase)
     }
 
-    pub(crate) fn query_cell_at_pos(
+    pub(crate) fn query_cell_at_pos<P: Phase>(
```

Comment:
It's not clear to me that this function will always return a cell belonging to a column at `phase`. Notice the path when `column_idx < self.columns.len()`, the `phase` parameter is ignored, and we don't know which is the phase of `self.columns[column_idx]`.

AFAIK, the cell manager here returns cells in this order (assume height = 3)

| Col1 | Col2 | Col3 |
| ---- | ---- | ---- |
| 0    | 3    | 6    |
| 1    | 4    | 7    |
| 2    | 5    | 8    |

The queries 0, 3 and 6 cause the `meta.advice_column_in(phase)`.   And the rest of the queries will reuse the column at whichever phase was chosen before.

I think the cell manager needs to be duplicated.  One cell manager for phase1, one cell manager for phase2.  Each one will create columns in the corresponding phase.

Reply:
> It's not clear to me that this function will always return a cell belonging to a column at `phase`. Notice the path when `column_idx < self.columns.len()`, the `phase` parameter is ignored, and we don't know which is the phase of `self.columns[column_idx]`.
> 
> AFAIK, the cell manager here returns cells in this order (assume height = 3)
> Col1 	Col2 	Col3
> 0 	3 	6
> 1 	4 	7
> 2 	5 	8
> 
> The queries 0, 3 and 6 cause the `meta.advice_column_in(phase)`. And the rest of the queries will reuse the column at whichever phase was chosen before.
> 
> I think the cell manager needs to be duplicated. One cell manager for phase1, one cell manager for phase2. Each one will create columns in the corresponding phase. And then cells for `data_rlcs` will be allocated with `cell_manager_phase2.query_cell(meta)`, where as the rest of the cells will be queried with `cell_manager_phase1.query_cell(meta)`.